public/app/features/dashboard-scene/settings/VariablesEditView.tsx

```diff
@@ -1,30 +1,145 @@
 import React from 'react';
 
 import { PageLayoutType } from '@grafana/data';
-import { SceneComponentProps, SceneObjectBase } from '@grafana/scenes';
+import { SceneComponentProps, SceneObjectBase, SceneObjectRef, SceneVariables, sceneGraph } from '@grafana/scenes';
 import { Page } from 'app/core/components/Page/Page';
 
+import { DashboardScene } from '../scene/DashboardScene';
 import { NavToolbarActions } from '../scene/NavToolbarActions';
-import { getDashboardSceneFor } from '../utils/utils';
 
 import { DashboardEditView, DashboardEditViewState, useDashboardEditPageNav } from './utils';
-
-export interface VariablesEditViewState extends DashboardEditViewState {}
+import { VariableEditorList } from './variables/VariableEditorList';
+export interface VariablesEditViewState extends DashboardEditViewState {
+  dashboardRef: SceneObjectRef<DashboardScene>;
+}
 
 export class VariablesEditView extends SceneObjectBase<VariablesEditViewState> implements DashboardEditView {
+  public static Component = VariableEditorSettingsListView;
+
   public getUrlKey(): string {
     return 'variables';
   }
 
-  static Component = ({ model }: SceneComponentProps<VariablesEditView>) => {
-    const dashboard = getDashboardSceneFor(model);
-    const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+  private get _dashboard(): DashboardScene {
+    return this.state.dashboardRef.resolve();
+  }
+
+  public getDashboard(): DashboardScene {
+    return this._dashboard;
+  }
+
+  private get _variablesScene(): SceneVariables {
+    return sceneGraph.getVariables(this.getDashboard());
+  }
+
+  public getVariablesScene(): SceneVariables {
+    return this._variablesScene;
+  }
+
+  private getVariableIndex = (identifier: string) => {
+    const variables = this.getVariables();
+    return variables.findIndex((variable) => variable.state.name === identifier);
+  };
+
+  public onDelete = (identifier: string) => {
+    // Find the index of the variable to be deleted
+    const variableIndex = this.getVariableIndex(identifier);
+    const { variables } = this._variablesScene.state;
+    if (variableIndex === -1) {
+      // Handle the case where the variable is not found
+      console.error('Variable not found');
+      return;
+    }
+
+    // Create a new array excluding the variable to be deleted
+    const updatedVariables = [...variables.slice(0, variableIndex), ...variables.slice(variableIndex + 1)];
+
+    // Update the state or the variables array
+    this._variablesScene.setState({ variables: updatedVariables });
+  };
+
+  public getVariables = () => {
+    return this._variablesScene.state.variables;
+  };
 
-    return (
-      <Page navModel={navModel} pageNav={pageNav} layout={PageLayoutType.Standard}>
-        <NavToolbarActions dashboard={dashboard} />
-        <div>variables todo</div>
-      </Page>
-    );
+  public getVariablesList = () => {
+    return this._variablesScene?.useState().variables;
   };
```

Comment:
never use a useState function like this, only allowed to use it from the react component.